content/post/2025/02-riscv-formal-extension-for-vc-formal.md

```diff
@@ -0,0 +1,298 @@
+
+Imagine deploying a RISC-V processor in an embedded system, only to discover an instruction-level bug after manufacturing—this could be a costly mistake. Formal verification ensures that such issues never make it past the design phase, and tools like **riscv-formal** help automate this process. However, many engineers rely on proprietary tools like **Synopsys VC Formal**, making integration with open-source solutions a challenge—until now.  
+
+The **riscv-formal** framework has been a game-changer, providing an open-source solution for verifying RISC-V cores against the official ISA specification. While powerful, its integration with industry-grade proprietary tools has remained complex. This project bridges that gap by extending **riscv-formal** to seamlessly interface with **Synopsys VC Formal**, enabling engineers to harness its advanced capabilities while maintaining a structured and efficient verification approach.  
+
+In this post, we’ll explore the role of formal verification in RISC-V core validation, the motivation behind this integration, and the methodology that makes it possible. Ultimately, this extension enhances accessibility and usability, providing the broader verification community with a streamlined, industry-compatible solution.  
+
+## About the Framework  
+
+### Before We Dive In...  
+Before diving into the project, let's take a quick look at the **riscv-formal** framework and understand why it plays a crucial role in RISC-V core verification.  
+
+### What is `riscv-formal`?  
+`riscv-formal` is an **open-source formal verification framework** designed to rigorously verify **RISC-V processors** against the **RISC-V ISA specification**. Built on the **FOSS SymbiYosys Formal Verification Flow**, it ensures that a processor implementation behaves correctly at the instruction and architectural levels.  
+
+### Why is `riscv-formal` Important?  
+- **Automated Property Checking:** Ensures compliance with the RISC-V ISA without requiring manual testbenches.  
+- **Minimal Setup, Maximum Coverage:** Works with any RISC-V design using an **RVFI (RISC-V Formal Interface)** wrapper.  
+
+### How it Works  
+1. **RVFI Wrapper**: Wraps the core-under-test with an **RVFI-compliant interface** for compatibility.  
+2. **Formal Checkers**: A set of **predefined checkers** checks the whole core thoroughly.  
+3. **Verification Engines**: Uses predefined **assertions and constraints** to validate core behavior.  
+4. **Counterexample Analysis**: If a check fails, it generates a **trace of execution (CEX)** to debug violations.  
+
+### Limitations & Extending to Proprietary Tools  
+While `riscv-formal` is highly effective for **open-source formal verification**, its native tooling (SBY/Yosys) **does not support proprietary EDA tools**. This is where our **extension** comes in—adapting `riscv-formal` for **Synopsys VC Formal**, bridging the gap between **open-source and commercial verification flows**.  
```

Comment:
The last sentence here is almost exactly the same as the first sentence of the next section.  I would suggest either dropping it from the next section and just start with
> By adapting the `riscv-formal` framework...

Or adjusting this section to just talking about the limitations and a very brief mention at the end that this is what your project aims to solve (using the word project here to link it to the next section of "Project Goals".